File: RxJava_Exam/Java/app/src/main/java/com/example/dgsw/rxjava_exam/MainActivity.java
package com.example.dgsw.rxjava_exam;

import android.os.Bundle;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.room.Room;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText mTodoEditText = findViewById(R.id.todo_edit);
        TextView mResultTextView = findViewById(R.id.result_text);

        AppDatabase db = Room.databaseBuilder(this, AppDatabase.class, "todo-db")
                .build();

        db.todoDao().getAll().observe(this, it -> mResultTextView.setText(it.toString()));

        findViewById(R.id.add_button).setOnClickListener(v -> {
            db.todoDao().insert(new Todo(mTodoEditText.getText().toString()))
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribe();
        });
    }
}

File: RxJava_Exam/Kotlin/app/src/main/java/com/example/dgsw/rxjava_exam/MainActivity.kt
package com.example.dgsw.rxjava_exam

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.room.Room
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.schedulers.Schedulers
import io.reactivex.schedulers.Schedulers.io

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val editText = findViewById<EditText>(R.id.todo_edit)
        val textView = findViewById<TextView>(R.id.result_text)
        val button = findViewById<Button>(R.id.add_button)

        val db = Room.databaseBuilder(
                applicationContext,
                AppDatabase::class.java,
                "todo-db-kt-RxJava"
        ).build()

        db.todoDao().getAll().observe (this, Observer { todo ->
                    textView.text = todo.toString()
                }
        ) // LiveData 사용

        button.setOnClickListener {
            db.todoDao().insert(Todo(editText.text.toString()))
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribe() //
        }
    }
}